### ISSUE TITLE
Document when to use?

### ISSUE THREAD

**Issue by jhabdas** [localheinz/github-changelog#158]
I've been watching this project for a few months because it looked interesting and I like the developer 😉 . Curious if it would make sense to add the use cases to the readme so others know when to use it versus something like [`standard-version`](https://github.com/conventional-changelog/standard-version)?

Not sure. But I can see from the change log the formatting here seems to fit better with the WP way of documenting changelog updates in humanese for end-users??

**Comment by localheinz** [comment]
@jhabdas 

Thank you for opening the issue - this sounds like a good idea.

Let me take a look at the linked package later, and then I will update `README.md`.

Also, I have been thinking, maybe it makes sense to allow to group pull requests by prefix?

That is, if people used

* `Major: `
* `Minor: `
* `Patch: `

as prefixes for pull request titles, it would be trivial to group the pull requests in a similar fashion as one would expect, for example, from http://keepachangelog.com/en/1.0.0/.

The primary idea of this package was to at least have the possibility to *generate* a changelog without the need to *manually write* it. If some discipline is applied when creating and naming pull requests, this is definitely better than nothing, I think.

What do you think?

**Comment by jhabdas** [comment]
Love that changelog resource. Awesome to see they're on 1.0 now. Yay! 😺 

Seems the pull request deal could be useful if people can remember to use it, and use it in accordance with SemVer. For my money what I do is let pull requests become a messy workspace for my contributors and then squash and merge their commits and update the commit messages using a `rebase -i` before moving code into master/trunk as a single commit with the commit message being the canonical place to do the versioning.

For example, someone adds a pull and I guide them through the PR and let the commits pile up. Then, before the merge I'll do a:

    git rebase -i head~20

And pick all the commits to squash, then update the commit message to something like:

```
feat($scope): add ability to do something awesome

This is the long description for do something awesome... It's long and stuff.

BREAKING CHANGE: Whoa, nelly. You're gonna have to refactor to upgrade. Here's how…

Closes \#40
```

When the rebase is complete the following happens when I merge that to master and run a release (using `standard-version`, in this case, and assuming the squashed commit is pushed back up to the `origin/topic` before hitting `origin/master`):

- Pull request closed automatically by the squashed commit appearing in `origin/master`
- Open issue \#40 related to the pull is closed automatically by commit message
- Release process updates the package manifest with a MAJOR version bumb (due to "BREAKING CHANGE")
- Existing CHANGELOG updated automatically with links to issues and pulls
- Git version is tagged and, optionally, NPM/whatever package is updated
- Release is sent up and most of all manual tasks are reduced to 1-3 programmatic expressions

Here are some examples where I've been using this approach:

- [Hyperdrive](https://github.com/comfusion/) – WordPress Beta Plugin
- [After Dark](https://github.com/comfusion/after-dark) – Hacker theme for Hugo

Not saying this is the way you'd want to go. WordPress and I believe the keepachangelog standard proselytize something very human readable IIRC, which means a Human readable changelog for wordpress still has a very legit use case which needs to be met by the community.

Phew! That got a little longer than I wanted. Hope it doesn't cause a headache. D:
